individual/Core/TaxLogic/Generic.php

```diff
@@ -37,14 +37,4 @@ public function setDeductions(fMoney $deductions)
 	{
 		$this->deductions = $deductions;
 	}
-
-	public function setTaxMode($mode)

```

Comment:
I have an intense problem with critical validation code being removed wholesale from the generic class.

You absolutely must change this into a virtual method, required by all derived classes. This is a critical change that must be done for project completion.